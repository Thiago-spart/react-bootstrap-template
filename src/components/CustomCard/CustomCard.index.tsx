import type { CustomCardProps } from "./CustomCard.types"

export const CustomCard: React.FC<CustomCardProps> = ({ bodyContent, link, title }) => {
	return (
		<div className="card" style={{ width: "18rem" }}>
			{/* <img className="card-img-top" src="..." alt="Card image cap" /> */}
			<div className="card-body">
				<h5 className="card-title">{title}</h5>
				<p className="card-text">{bodyContent}</p>
				<a href={link} className="btn btn-primary">Go somewhere</a>
			</div>
		</div>
	)
}