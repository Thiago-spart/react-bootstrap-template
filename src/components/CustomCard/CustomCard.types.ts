import { ReactNode } from "react";

export interface CustomCardProps {
	title: string;
	bodyContent: ReactNode;
	link: string
}