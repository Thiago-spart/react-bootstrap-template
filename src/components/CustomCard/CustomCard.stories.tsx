/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { CustomCard } from "./CustomCard.index";

const meta: Meta<typeof CustomCard> = {
	title: "Componentes/Button",
	component: CustomCard,
};

const Template: ComponentStory<typeof CustomCard> = props => <CustomCard {...props} />;

export const Default = Template.bind({});
Default.args = {
	title: "Titulo",
	bodyContent: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
	link: "https://www.google.com"
};

export default meta;
