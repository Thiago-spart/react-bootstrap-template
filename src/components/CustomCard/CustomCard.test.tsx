import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";

import * as stories from "./CustomCard.stories";

const { Default } = composeStories(stories);

describe("CustomCard component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});

	it("should be clicked", () => {
		render(<Default />);

		expect(screen.getByText("Titulo")).toBeTruthy();
	});
});
