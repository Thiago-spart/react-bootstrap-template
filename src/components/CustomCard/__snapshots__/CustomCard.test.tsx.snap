// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCard component should match snapshot 1`] = `
<div>
  <div
    class="card"
    style="width: 18rem;"
  >
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Titulo
      </h5>
      <p
        class="card-text"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <a
        class="btn btn-primary"
        href="https://www.google.com"
      >
        Go somewhere
      </a>
    </div>
  </div>
</div>
`;
